### ISSUE TITLE
Utility functions should be scoped into specific files and exported

### ISSUE THREAD

**Issue by backnight** [strapi/strapi#4007]
**Describe the topic**
Utility functions should be scoped into specific files and exported.

**The problem**

I just wanted to customize a simple search: (params) function in one of my api services.
I needed to do the same thing as Strapi and add a join() but as example, Strapi doesn't export the buildSearchQuery function in https://github.com/strapi/strapi/blob/master/packages/strapi-hook-bookshelf/lib/queries.js#L491

So basically, I have to copy paste all the code to be able to re-do the same logic.

**Comment by lauriejim** [comment]
Hello @backnight ! Thank you for this message.
Ping @alexandrebodin  - this is the guy to talk with you about that.

**Comment by alexandrebodin** [comment]
@backnight Thank you for this comment. To give you more context there are two main reasons these functions aren't exposed.

- Strapi does not have stable apis yet. We are still moving a lot of things in the database layer and will be doing refactoring again quite a lot i the coming months. So creating public apis that change every week is really not our intention.
- These functions are really only internal functions of the library and aren't intended to be exposed anytime soon. This is the kind of utility taht will stay internal so we can change it anytime without impacting the users. We need to keep some room for modifications, and increasing api surface area will make the project impossible to maintain.

I'm closing this issue as there are no actions to do. I will keep in mind that usecase for future developments !
